examples/glue_benchmark/glue.py

```diff
@@ -112,200 +69,127 @@ def split_features(x):
+    def preprocess_fn(feature, label):
+        return preprocessor(feature), label
 
-def preprocess_data(preprocess_fn, dataset):
-    """Run `proprocess_fn` on input dataset then batch & prefetch."""
     return (
         dataset.map(preprocess_fn)
         .batch(FLAGS.batch_size)
         .prefetch(tf.data.AUTOTUNE)
     )
 
 
-def generate_submission_files(finetuning_model, test_ds, idx_order):
-    """Generate GLUE leaderboard submission files."""
-    filenames = {
-        "cola": "CoLA.tsv",
-        "sst2": "SST-2.tsv",
-        "mrpc": "MRPC.tsv",
-        "qqp": "QQP.tsv",
-        "stsb": "STS-B.tsv",
-        "mnli_matched": "MNLI-m.tsv",
-        "mnli_mismatched": "MNLI-mm.tsv",
-        "qnli": "QNLI.tsv",
-        "rte": "RTE.tsv",
-        "wnli": "WNLI.tsv",
-        "ax": "AX.tsv",
-    }
-
-    labelnames = {
-        "mnli_matched": ["entailment", "neutral", "contradiction"],
-        "mnli_mismatched": ["entailment", "neutral", "contradiction"],
-        "ax": ["entailment", "neutral", "contradiction"],
-        "qnli": ["entailment", "not_entailment"],
-        "rte": ["entailment", "not_entailment"],
-    }
-    if not os.path.exists(FLAGS.submission_directory):
-        os.makedirs(FLAGS.submission_directory)
-    filename = FLAGS.submission_directory + "/" + filenames[FLAGS.task_name]
-    labelname = labelnames.get(FLAGS.task_name)
-
-    predictions = finetuning_model.predict(test_ds)
-    if FLAGS.task_name == "stsb":
-        predictions = np.squeeze(predictions)
-    else:
-        predictions = np.argmax(predictions, -1)
-
-    # Map the predictions to the right index order.
-    idx_order = list(idx_order.as_numpy_iterator())
-    contents = ["" for _ in idx_order]
-    for idx, pred in zip(idx_order, predictions):
-        if labelname:
-            pred_value = labelname[int(pred)]
-        else:
-            pred_value = pred
-            if FLAGS.task_name == "stsb":
-                pred_value = min(pred_value, 5)
-                pred_value = max(pred_value, 0)
-                pred_value = f"{pred_value:.3f}"
-        contents[idx] = pred_value
-
-    with tf.io.gfile.GFile(filename, "w") as f:
-        # GLUE requires a format of index + tab + prediction.
-        writer = csv.writer(f, delimiter="\t")
-        # Write the required headline for GLUE.
-        writer.writerow(["index", "prediction"])
-
-        for idx, value in enumerate(contents):
-            writer.writerow([idx, value])
-
-
-def connect_to_tpu(tpu_name):
-    resolver = tf.distribute.cluster_resolver.TPUClusterResolver.connect(
-        tpu=tpu_name
+def main(_):
+    warnings.warn("DISCLAIMER:This script only supports GLUE/mrpc (for now).")
+    # checking task version (erroring out other testes except "mrpc")
+    if FLAGS.task != "mrpc":
+        raise ValueError("task - mrpc is only supported currently.")
+
+    print(tf.__version__)
+    print("GPU available : ", tf.test.is_gpu_available())
+
+    print("=" * 120)
+    print(
```

Comment:
Let's use `logging` instead of `print` so that this script is easier to run on cloud.

`from absl import logging`, then `logging.info()`